File: python/extract_doc.py
#!/usr/local/bin/python

'''
This script performs a synchronous extraction of the supplied PDF and is limited to
PDFs of ~4.5MB or less with an extraction runtime under 30s (extractions rarely take
longer than 30s unless they require OCR). See https://docs.sensible.so/docs/api-tutorial-sync
for more details
'''

import requests
import json
import os

# The name of a document type in Sensible, e.g., auto_insurance_quote 
doc_type = "auto_insurance_quote" # YOUR_DOCUMENT_TYPE TODO
# The path to the PDF you'd like to parse
doc_local_path = "../TODELETE_auto_insurance_anyco.pdf" # YOUR_PDF.pdf
# Your Sensible API key
API_KEY = "YOUR_API_KEY"

def extract_doc():  
  with open(doc_local_path, 'rb') as pdf_file:
      pdf_bytes = pdf_file.read()
  payload = pdf_bytes
  headers = {
  'Authorization': 'Bearer {}'.format(API_KEY),
  'Content-Type': 'application/pdf'
}
  response = requests.request("POST", "https://api.sensible.so/dev/extract/{}".format(doc_type), headers=headers, data=payload)
  try:
    response.raise_for_status()
  except requests.RequestException as err:
    print(response.text)
  print(json.dumps(response.json(), indent=2))

if __name__ == '__main__':
  extract_doc()

File: node/extract-doc.js
#!/usr/bin/env node

// This script performs a synchronous extraction of the supplied PDF and is limited to
// PDFs of ~4.5MB or less with an extraction runtime under 30s (extractions rarely take
// longer than 30s unless they require OCR). See https://docs.sensible.so/docs/api-tutorial-sync
// for more details

const fs = require("fs");
const fetch = require("isomorphic-fetch");

// The name of a document type in Sensible, e.g., auto_insurance_quote
const DOCUMENT_TYPE = "YOUR_DOCUMENT_TYPE";

// The path to the PDF you'd like to parse
// If the PDF is over ~4.5MB use the async-extract-doc.js script
const DOCUMENT_PATH = "YOUR_PDF.pdf";

// Your Sensible API key
const API_KEY = "YOUR_API_KEY";

async function main() {
  const headers = new Headers();
  headers.append("Authorization", `Bearer ${API_KEY}`);
  headers.append("Content-Type", "application/pdf");

  const body = fs.readFileSync(DOCUMENT_PATH);

  // TODO: Replace dev with v0
  const response = await fetch(
    `https://api.sensible.so/dev/extract/${DOCUMENT_TYPE}`,
    {
      method: "POST",
      headers,
      body,
    }
  );

  if (!response.ok) {
    console.log(await response.text());
  } else {
    console.log(JSON.stringify(await response.json(), null, 2));
  }
}

main();
